Error compiling WRF Plus in WRF 4.0.2 on module_wrf_top.f90

I keep getting an error message when I try to compile WRF Plus.
I configured with intel compilers (option 10) and netcdf 4.6.2 (without netcdf4), and I have no problem when I compiled WRF and WPS.
So, when I compile WRF Plus I get the following error messages:

```
mpiifort -o ../main/module_wrf_top.o -c -O3 -xCORE-AVX512 -w  -auto -ftz -fno-alias -fp-model fast=1 -no-prec-div -no-prec-sqrt -FR -convert big_endian -auto -align array64byte    -I../dyn_em -I../dyn_nmm  -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/external/esmf_time_f90  -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/main -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/external/io_netcdf -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/external/io_int -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/frame -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/share -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/phys -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/wrftladj -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/chem -I/home/cross/WRF/4.0.2_nc3/WRF_Plus/inc -I/home/cross/Software/nc3/libs/netcdf/include  -r8        -real-size `expr 8 \* 8` -i4  ../main/module_wrf_top.f90
../main/module_wrf_top.f90(756): error #6099: An ENDDO statement occurred without a corresponding DO or DO WHILE statement.
   ENDDO
^
compilation aborted for ../main/module_wrf_top.f90 (code 1)
../configure.wrf:346: recipe for target '../main/module_wrf_top.o' failed
```

So I've searched this problem, and it supposed to be some ENDEIF missing, but it doesn't seems to be that in this case. here is the code from module_wrf_top.f90 from line 745 o 756

```
IF ( config_flags%check_TL .or. config_flags%check_AD ) THEN

   CALL allocate_grid ( )

   !$OMP PARALLEL DO    &
   !$OMP DEFAULT (SHARED) PRIVATE ( ij ) &
   DO ij = 1 , head_grid%num_tiles
      CALL copy_grid_to_s ( head_grid , &
                            head_grid%i_start(ij), head_grid%i_end(ij),                &
                            head_grid%j_start(ij), head_grid%j_end(ij)                 )
   ENDDO
```
does anyone know what could be the problem here?

--- a/main/module_wrf_top.F
+++ b/main/module_wrf_top.F
@@ -872,13 +872,14 @@ IF ( config_flags%check_TL .or. config_flags%check_AD ) THEN
    CALL allocate_grid ( )
 
    !$OMP PARALLEL DO    &
-   !$OMP DEFAULT (SHARED) PRIVATE ( ij ) &           
+   !$OMP DEFAULT (SHARED) PRIVATE ( ij )           
    DO ij = 1 , head_grid%num_tiles
       CALL copy_grid_to_s ( head_grid , &
                             head_grid%i_start(ij), head_grid%i_end(ij),                &
                             head_grid%j_start(ij), head_grid%j_end(ij)                 )
    ENDDO
-
+   !$OMP END PARALLEL DO
+   
    CALL       wrf_message ( "wrf: calling nonlinear integrate" )
    model_config_rec%dyn_opt = dyn_em
 

